wordpress-mobile/gutenberg-mobile
# Update caret position after inserting link

To Fix: wordpress-mobile/gutenberg-mobile#679

Gutenberg PR: https://github.com/WordPress/gutenberg/pull/14317

With this PR we are putting the caret at the end of the linked text when we insert a link. This needs to be tested both on Android & iOS.

Following screen captures are taken from this PRs branch:

iOS:

![link-ios](https://user-images.githubusercontent.com/5032900/53969723-58fd8980-410a-11e9-9497-e41bcf8c0eef.gif)

Android:

![link-android](https://user-images.githubusercontent.com/5032900/53969735-5c911080-410a-11e9-8c9d-0125fb909adc.gif)

**TO TEST**
Testing prerequisites

For WPiOS

Checkout the PRs branch to any arbitrary folder and cd .. to it
run yarn install, yarn start
Open XCode WPiOS on the latest develop
Clean build folder on Xcode, and then run the app

For WPAndroid

open grade.properties at WordPress-Android folder
add wp.BUILD_GUTENBERG_FROM_SOURCE = true to grade.properties
checkout the PRs branch in the subrepo of WordPress-Android repo
cd to WordPress-Android/libs/gutenberg-mobile
run yarn install, yarn start
yarn wpandroid on a separate terminal in the same directory

Test steps:

Test 1 - Inserting link when no text is selected

- Tap somewhere in a paragraph/heading block
- Tap links button in the toolbar
- Add a link and close the links ui
- Verify that the caret is at the end of the inserted link

Test 2 - Editing an existing link

- Select an existing link in a paragraph/heading block
- Tap links button in the toolbar
- Update the visual text of the link and close the links ui
- Verify that the caret is at the end of the updated link

Test 3 - Select a text and add a link

- Select some text in a paragraph/heading block
- Tap links button in the toolbar
- Add a url for the link and close the links ui
- Verify that the selection remains same
